### PR TITLE
gateway: package reogranization

### PR DESCRIPTION
**Pull request by bobheadxi**
:tickets: **Ticket(s)**: n/a

---

## :construction_worker: Changes

Since package `api` is at risk of getting crowded, making this change before the semester starts:

- user stuff now in package `api/user`
- club stuff now in package `api/club`

## :flashlight: Testing Instructions

tests should pass as before

### CODE REVIEW

File: gateway/api/club/club.go [hunk 1]
```diff
@@ -14,15 +14,16 @@ import (
...
+type Router struct {
 	l   *zap.SugaredLogger
 	c   pinpoint.CoreClient
 	mux *chi.Mux
 }
 
-func newClubRouter(l *zap.SugaredLogger, core pinpoint.CoreClient) *ClubRouter {
-	c := &ClubRouter{l.Named("clubs"), core, chi.NewRouter()}
+// NewClubRouter instantiates a new router for club functionality
+func NewClubRouter(l *zap.SugaredLogger, core pinpoint.CoreClient) *Router {
```
Comment (leesw98) [inline gateway/api/club/club.go]: Wouldn't `NewRouter` be a better name for this?

File: gateway/api/user/user.go [hunk 2]
```diff
@@ -12,15 +12,16 @@ import (
...
+type Router struct {
 	l   *zap.SugaredLogger
 	c   pinpoint.CoreClient
 	mux *chi.Mux
 }
 
-func newUserRouter(l *zap.SugaredLogger, core pinpoint.CoreClient) *UserRouter {
-	u := &UserRouter{l.Named("users"), core, chi.NewRouter()}
+// NewUserRouter instantiates a new router for handling user functionality
+func NewUserRouter(l *zap.SugaredLogger, core pinpoint.CoreClient) *Router {
```
Comment (leesw98) [inline gateway/api/user/user.go]: Same here

**Review by bwdmonkey — APPROVED** [review]

File: gateway/api/club/club.go (hunk 1 above)
Comment (leesw98) [reply gateway/api/club/club.go]: Up to you

Comment (bobheadxi) [reply gateway/api/club/club.go]: its okay  i wouldnt want to confuse you tom tom

**Pull request merged**